edgelet/iotedge-proxy/src/signal.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use futures::Future;
+
+pub type ShutdownSignal = Box<dyn Future<Item = (), Error = ()> + Send>;
+
+pub fn shutdown() -> ShutdownSignal {
+    imp::shutdown()
+}
+
+#[cfg(unix)]
+mod imp {
+    use std::fmt;
+
+    use futures::{future, Future, Stream};
+    use log::info;
+    use tokio_signal::unix::{Signal, SIGINT, SIGTERM};
+
+    use super::ShutdownSignal;
+
+    pub(super) fn shutdown() -> ShutdownSignal {
+        let signals = [SIGINT, SIGTERM].iter().map(|&sig| {
+            Signal::new(sig)
+                .flatten_stream()
+                .into_future()
+                .map(move |_| {
+                    info!(
+                        target: "iotedged::signal",
+                        "Received {}, starting shutdown",
+                        DisplaySignal(sig),
+                    );
+                })
+        });
+        let on_any_signal = future::select_all(signals)
+            .map(|_| ())
+            .map_err(|_| unreachable!("Signal never returns an error"));
+        Box::new(on_any_signal)
+    }
+
+    #[derive(Clone, Copy)]
+    struct DisplaySignal(i32);
+
+    impl fmt::Display for DisplaySignal {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            let s = match self.0 {
+                SIGINT => "SIGINT",
+                SIGTERM => "SIGTERM",
+                other => return write!(f, "signal {}", other),
+            };
+            f.write_str(s)
+        }
+    }
+}
+
+#[cfg(not(unix))]
```

Comment:
Did you test this on Windows? I have had problems in the past when iotedge was running as a Windows Service.

Reply:
We are not targeting windows for k8s work so far. I added this code just to make it compile in windows along with other parts of edgelet. Should try it though